### DISCUSSION TITLE
[Announcement] The next major version is under alpha test

### DISCUSSION THREAD

**Discussion by windingwind** [windingwind/zotero-better-notes#429] [Announcements]
Hi,

I am the developer of this plugin. Thanks for your support of my work!

I am currently working on the migration for the coming Zotero 7. During the migration:

- All issues will be suspended and not replied to (unless they are urgent);
- We still accept PRs, but please let me know what you are going to do before you start working on the PR;

The migration is expected to last for at least a month, starting from February 6, 2023.

Thanks for your understanding!

Best,
windingwind

**Comment by windingwind** [comment]
It's almost a total rewrite... Took longer than I expected and I still need some time to make the first alpha available.

I would be very happy if someone are willing to join and work with me... But currently, I have to do it on my own.

Thanks for your patience!

**Comment by windingwind** [comment]
Glad to announce to all of you that the first v1.0.0 alpha is finally there! See https://github.com/windingwind/zotero-better-notes/releases/tag/1.0.0-0 for installation.

There have been very, very fundamental changes to this plugin. Most of the features we have in the previous versions are kept to the most. However, due to API changes, most of the note templates may not work for the coming new version anymore. The good news is: you can now write note templates with MarkDown and preview them in real time.

Note that there will be some differences between the alpha/beta version and the final releases:

- alpha/beta versions always pop up a hint when syncing is triggered.
- alpha/beta versions may use more resources(memory, CPU).
- there are no ensurances to your data security with alpha/beta versions.

We strongly suggest you create a new profile for testing the alpha version of this plugin.

Welcome any suggestions and bug reports!

**Comment by lil-i321** [comment]
For 1.0.0-3：不能同步，且不能生成md文件,

![Snipaste_2023-04-15_15-51-28](https://user-images.githubusercontent.com/57334649/232197343-64f2045d-30b7-4eb5-9800-9c30ee21b901.png)

![Snipaste_2023-04-15_15-49-25](https://user-images.githubusercontent.com/57334649/232197261-9fe1052f-d643-4b40-a41a-3bcc59b9282e.png)

**Comment by lil-i321** [comment]
For 1.0.0-3：
1、不能实现 插入模板到工作区笔记，同样的自制模板0.8.9可以插入。
2、插入自带模板 报错。
![Snipaste_2023-04-15_19-52-18](https://user-images.githubusercontent.com/57334649/232217922-bc3963f3-6120-40e1-a689-56eb0278238d.png)

**Comment by windingwind** [comment]
@lil-i321 
> For 1.0.0-3：不能同步，且不能生成md文件,

fixed in v1.0.0-4

> For 1.0.0-3：
1、不能实现 插入模板到工作区笔记，同样的自制模板0.8.9可以插入。
2、插入自带模板 报错。

> However, due to API changes, most of the note templates may not work for the coming new version anymore.

**Comment by windingwind** [comment]
@lil-i321 
> 插入自带模板 报错。

fixed in 1.0.0-6

**Comment by lil-i321** [comment]
> 

@windingwind 
谢谢更新！作者大大辛苦！
For 1.0.0-6：
1、安装插件后，不会弹出 “重启”提示，（试了一次）；
2、可以导出md文件，但字数多的笔记，导出成功率不高，依然会 “未响应”；
3、关于同步，字数少的笔记，可以实现双向同步更新，但是，字数多的笔记，只要点击更新，就会“未响应”，100%；
4、插入自带模板 不报错了，但不显示内容，为空。

**Comment by lil-i321** [comment]
> 

@windingwind 
3、关于同步，字数少的笔记，可以实现双向同步更新，但是，字数多的笔记，只要点击更新，必须等待，不能有任何操作，否则必然“未响应”，等待3min，可以实现双向同步更新，（3500字的笔记）。

**Reply by windingwind** [reply]
可能是电脑配置的缘故？我测试了3万字的笔记导出，会略卡5s左右，3千字的笔记导出并设置双向同步（带有50条不同子笔记链接），几乎是瞬间完成

**Reply by lil-i321** [reply]
应该是那个笔记的问题，来回导出几次，这个笔记直接消失了，还好有导出的md文件，不然就哭了。。。
现在没问题了。

**Comment by windingwind** [comment]
> For 1.0.0-6：
1、安装插件后，不会弹出 “重启”提示，（试了一次）；

现在插件安装/更新后不需要重启。

> 2、可以导出md文件，但字数多的笔记，导出成功率不高，依然会 “未响应”；
3、关于同步，字数少的笔记，可以实现双向同步更新，但是，字数多的笔记，只要点击更新，就会“未响应”，100%；

烦请提供一个导出失败的笔记样例/md样例，我测试一下

> 4、插入自带模板 不报错了，但不显示内容，为空。

请确认操作流程。菜单栏中的是“插入到工作区笔记”，要插入在当前笔记内，请从笔记编辑器上方齿轮按钮菜单选择 插入模板到光标。

新版本实际上不包含任何Item或text类的自带模板，以前版本的模板可能会需要更改。

**Reply by lil-i321** [reply]
3、重新做了个笔记，49k，远大于之前的，没有问题，秒同步秒导出，可能是之前的笔记有问题吧，
4、按照这个操作流程，自己的模板不用修改就可以成功插入。
感谢作者大大！

**Reply by lil-i321** [reply]
@windingwind 
这个md文件只有2条内容。
先添加第一条注释，可以正常导出，md文件中看到的zotero链接也是正常长度；
再添加第二条注释，导出时就会非常卡，等待30s后，也能生成md文件，但是zotero链接就会非常长，并且第一条的zotero链接也变得非常长。
我怀疑，上次我提的问题，就是这个原因：某一条注释在导出的过程中**链接异常变长，并使所有链接都变得非常长**，导致整个文件字符超级多，以至于 导出/同步时 卡死。
**为什么有些注释的链接会超级长？zotreo双链能否控制导出长度？**

[笔记：A007-N4PLNA5G.md](https://github.com/windingwind/zotero-better-notes/files/11263351/A007-N4PLNA5G.md)

**Comment by lil-i321** [comment]
for 1.0.0-6
### 字体颜色不能正常导出到md文件，
笔记中只有链接内的文字可以导出颜色，链接外的注释文字和手输文字都不可以，
高亮文字的颜色可以导出。

![Snipaste_2023-04-17_23-45-11](https://user-images.githubusercontent.com/57334649/232541601-b1648189-91ac-45a5-afc8-0dd6218db571.png)

**Comment by lil-i321** [comment]
作者大大能解答一下这两个问题吗？

**Reply by windingwind** [reply]
我最近在出差，还没时间看，不好意思

**Reply by lil-i321** [reply]
抱歉了，我还以为是我这不是啥问题呢，，，

**Reply by windingwind** [reply]
要解决得花时间改代码呀

**Reply by lil-i321** [reply]
明白明白，插件已然是生产力工具了，感谢！

**Comment by ippreacher** [comment]
笔记工作区的标签页上点右键，无法“移动到新窗口”，不知是否我这里的问题，希望能解决，感谢！

**Reply by windingwind** [reply]
已实现

**Comment by Paulgudring** [comment]
thanks for your contribution to the community! but after updating to 1.0.1, many template you have created became invaild. there's gonna be a lot of sequent work to do... like #456

> However, due to API changes, most of the note templates may not work for the coming new version anymore.

would you conider providing paradigm for templates later like before?

![image](https://github.com/windingwind/zotero-better-notes/assets/81338177/c35af80a-86c1-4b0c-9c68-22202c0dd7e6) 

collecting note by color helps me a lot.

**Reply by windingwind** [reply]
Yes. I will write a new documentation for developers. The new update brings many standard APIs, so most of the old templates that use some inner APIs in the old version will no longer work.

Writing a template would be significantly easier than before. It now supports Markdown and preview, without the need to test the embedded code yourself.

**Comment by ShangkunLi** [comment]
想请问一下，在better note中如何使用“OCR math from image annotations”，我点击了这个latex按钮之后没有反应，想请问一下是什么问题呢？
<img width="209" alt="Screenshot 2023-07-05 at 14 25 43" src="https://github.com/windingwind/zotero-better-notes/assets/95337731/0b445095-555d-4fd2-83a7-cbce671eafe3">

**Reply by windingwind** [reply]
这个功能还没有移植到新版本，正在考虑是不是要移除
